ipi/engine/forces.py

```diff
@@ -417,11 +429,30 @@ def extra_gather(self):
         """Obtains the extra string information for each replica.
 
         Returns:
-           A list of the extra strings of each replica of the system.
+           A list of the extra dictionaries for each replica of the system.
         """
 
         self.queue()
-        return [b.extra for b in self._forces]
+
+        # converts a list of dictionaries to a dictionary of lists
+        fc_extra = {}
+        for e in self._forces[0].extra:
+            fc_extra[e] = []
+
+        for b in self._forces:
+            for e in b.extra:
+                if not e in fc_extra:
+                    raise KeyError(
+                        "Extras mismatch between beads in the same force component, key: "
+                        + e
+                    )
+                fc_extra[e].append(b.extra[e])
+
+        # force_extras should be numerical, thus can be converted to numpy arrays
```

Comment:
"should be"? normally they are and that makes the most sense, but I don't see where the error is raised in a clean way. should be verbose.

Reply:
Counting on np.asarray() to raise an error.

Reply:
ok, let's raise an actual error to help the users...